model/buildlogger.go

```diff
@@ -196,7 +196,35 @@ func (l *Log) Append(ctx context.Context, lines []LogLine) error {
 		Start:    lines[0].Timestamp,
 		End:      lines[len(lines)-1].Timestamp,
 	}
-	return errors.Wrap(l.appendLogChunkInfo(ctx, info), "problem updating log metadata during upload")
+	if err = l.appendLogChunkInfo(ctx, info); err != nil {
+		return errors.Wrap(err, "problem updating log metadata during upload")
+	}
+
+	l.addToStatsCache(lines)
+
+	return nil
+}
+
+func (l *Log) addToStatsCache(lines []LogLine) {
+	linesLen := 0
+	for _, line := range lines {
+		linesLen += len(line.Data)
```

Comment:
Nit: `linesSize` or something to not confuse the reader with the number of log lines since `len` can be used to get the number lines in the `LogLine` slice.